openmdao/core/system.py

```diff
@@ -715,6 +717,612 @@ def _declare_options(self):
         """
         pass
 
+    def _have_output_solver_options_been_applied(self):
+        """
+        Check to see if the cached output solver options were applied.
+        """
+        been_applied = True
+        for subsys in [self] + self._subsystems_myproc:
+            if subsys._output_solver_options:  # If options dict not empty, has not been applied
+                been_applied = False
```

Comment:
You can probably put a break here to terminate the loop, once you find your first output_solver_options.

Reply:
Good catch. Fixed